config/config.go

```diff
@@ -76,11 +80,29 @@ func loadFromFile(filePath string, cfg *Config) error {
 	return err
 }
 
-func WithConfig(ctx context.Context, cfg *Config) context.Context {
-	return context.WithValue(ctx, configContextKey, cfg)
-}
+func setDefaults(cfg *Config) {
+	// Paths
+	if cfg.BinDir == "" {
+		cfg.BinDir = filepath.Join(cfg.HomeDir, "bin")
+	}
+	if cfg.RunDir == "" {
+		cfg.RunDir = filepath.Join(cfg.HomeDir, "run")
+	}
+	if cfg.VarDir == "" {
+		cfg.VarDir = filepath.Join(cfg.HomeDir, "var")
+	}
+
+	if cfg.HTTPPort < 1 {
+		cfg.HTTPPort = 4000
+	}
+
+	// Log
+	if cfg.Log.SyslogPort < 1 {
+		cfg.Log.SyslogPort = 4500
+	}
 
-func GetConfig(ctx context.Context) (*Config, bool) {
-	cfg, ok := ctx.Value(configContextKey).(*Config)
-	return cfg, ok
+	// GUI
+	if cfg.GUI.Port < 1 {
```

Comment:
Why < 1 (here and elsewhere?) are negative numbers possible or meaningful in any way?

Reply:
I think that I should change this to be `!= 0` and update the port types in config to `uint` instead.